src/puppetlabs/puppetdb/http/catalogs.clj

```diff
@@ -48,10 +48,12 @@
                       (str (:url-prefix globals))))
 
     [node "edges" &]
-    (comp (edges/edges-app version false) (partial http-q/restrict-query-to-node node))
+    (-> (comp (edges/edges-app version false) (partial http-q/restrict-query-to-node node))
+        (wrap-with-parent-check version :node node))
 
     [node "resources" &]
-    (comp (resources/resources-app version false) (partial http-q/restrict-query-to-node node))))
+    (-> (comp (resources/resources-app version false) (partial http-q/restrict-query-to-node node))
+        (wrap-with-parent-check version :node node))))
 
 (defn catalog-app
   [version]
```

Comment:
similar comment here as with facts. Now that we 404 in all cases we should change the existing catalogs error so the phrasing matches.

```
curl -X GET http://localhost:8080/pdb/query/v4/catalogs/foo
{
  "error" : "Could not find catalog for foo"
}
```
vs
```
curl -X GET http://localhost:880/pdb/query/v4/catalogs/foo/resources
{
  "error" : "No information is known about node foo"
}
```